keps/sig-api-machinery/2896-openapi-v3/README.md

```diff
@@ -0,0 +1,523 @@
+
+It is technically possible for the aggregator handling OpenAPI v3 to download from the OpenAPI v2 schema, but there are a few tradeoffs that I believe make this not worthwhile. The OpenAPI v3 aggregator must publish OpenAPI v3, and OpenAPI v2 will need to be converted if it is intended to be used. The v3 schema is designed to be a lossless representation of a user's spec, but by going through OpenAPI v2, critical information may have already been lost. Publishing an inaccurate v3 schema is not desirable and is not in line with the primary goal of this KEP. Converting from OpenAPI v2 to v3 via the Swagger/OpenAPI structs after publishing is also much more complicated than modifying the OpenAPI generation before publishing.
+
+By not publishing the v3 spec for old aggregated apiservers, the v3 spec is not a full representation of all the resources and endpoints available within the cluster. However, the scope of this is limited to select group-versions claimed by the select aggregated apiservers. A group-version in v3 will either have a fully lossless OpenAPI v3 or be non-existent (old aggregated apiserver), preventing a partial lossy spec from being published. This should also serve as a good incentive for users to upgrade their aggregated apiservers to the latest version and publish OpenAPI v3.
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+
+Consider the following in developing a test plan for this enhancement:
+- Will there be e2e and integration tests, in addition to unit tests?
+- How will it be tested in isolation vs with other components?
+
+No need to outline all of the test cases, just the general strategy. Anything
+that would count as tricky in the implementation, and anything particularly
+challenging to test, should be called out.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+For alpha, unit tests will be included to ensure that v3 schemas are properly generated and published. A validator will also be used to ensure that the spec generated is valid OpenAPI v3.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Native types are updated to capture capabilities introduced with v3
+- Definition names of native resources are updated to omit their package paths
+- Parameters are reused as components
```

Comment:
I would want `kubectl explain` here too as a beta step. It will prove that the spec is working fine.